lib/insights/api/common/open_api/generator.rb

```diff
@@ -562,6 +570,13 @@ def handle_custom_route_action(_route_action, _verb, _primary_collection)
           def schema_overrides
             {}
           end
+
+          def validate_operation_id(operation_id, klass_name)
+            if @operation_ids.include?(operation_id)
```

Comment:
@mkanoor Would we want to retain the operation_id and klass_name from the conflicting element?  Maybe instead of using an array we should use a hash to store the `operation_id` as the key and the `klass_name` as the value.

The check could become:
`@operation_ids.has_key?(operation_id)` 

Or we could just do the lookup and if the result is non-`nil` we report the `operation_id` and the previous and current `klass_name`.

```ruby
def validate_operation_id(operation_id, klass_name)
  previous_klass = @operation_ids[operation_id]
  if previous_klass
    raise ArgumentError, "operation id cannot be duplicates, #{operation_id} in class #{klass_name}  Previous class #{previous_klass}"
  end
  @operation_ids[operation_id] = klass_name
end

```

Reply:
@gmcculloug Please review